setup.py

```diff
@@ -74,53 +95,93 @@ def run_tests(self):
+            )
 
-        try:
-            cmake_command = os.environ.get("CMAKE_COMMAND", "cmake")
-            build_type = "RelWithDebInfo" if DEBUG_COMPILE else "Release"
-            opts = ["-DCMAKE_BUILD_TYPE={}".format(build_type)]
-            if platform.system() == "Windows":
-                opts.extend(["-A", "x64" if platform.architecture()[0] == "64bit" else "Win32"])
-            else:
-                opts.extend(["-G", "Ninja"])
-                ninja_command = os.environ.get("NINJA_COMMAND", "")
-                if ninja_command:
-                    opts.append("-DCMAKE_MAKE_PROGRAM={}".format(ninja_command))
-
-            for source_dir in to_build:
-                try:
-                    build_dir = tempfile.mkdtemp()
-                    subprocess.check_call([cmake_command, "-S", source_dir, "-B", build_dir] + opts)
-                    subprocess.check_call([cmake_command, "--build", build_dir, "--config", build_type])
-                finally:
-                    if not DEBUG_COMPILE:
-                        shutil.rmtree(build_dir, ignore_errors=True)
-
-            BuildExtCommand.build_extension(self, ext)
-        except Exception as e:
-            print('WARNING: building extension "%s" failed: %s' % (ext.name, e))
-            raise
+            try:
+                filename, http_response = urlretrieve(ddwaf_download_address, ddwaf_archive_name)
+            except HTTPError as e:
+                print("No archive found for dynamic library ddwaf : " + ddwaf_archive_dir)
+                raise e
+
+            # Open the tarfile first to get the files needed.
+            # This could be solved with "r:gz" mode, that allows random access
+            # but that approach does not work on Windows
+            with tarfile.open(filename, "r|gz", errorlevel=2) as tar:
+                dynfiles = [c for c in tar.getmembers() if c.name.endswith(SUFFIX)]
+
+            with tarfile.open(filename, "r|gz", errorlevel=2) as tar:
+                print("extracting files:", [c.name for c in dynfiles])
+                tar.extractall(members=dynfiles, path=HERE)
+
+                os.rename(os.path.join(HERE, ddwaf_archive_dir), arch_dir)
+                # cleaning unwanted files
+                tar.close()
+
+            # Rename ddwaf.xxx to libddwaf.xxx so the filename is the same for every OS
+            original_file = os.path.join(arch_dir, "lib", "ddwaf" + SUFFIX)
+            if os.path.exists(original_file):
+                renamed_file = os.path.join(arch_dir, "lib", "libddwaf" + SUFFIX)
+                os.rename(original_file, renamed_file)
+
+            os.remove(filename)
+
+    def run(self):
+        LibDDWaf_Download.download_dynamic_library()
+        BuildPyCommand.run(self)
+
+
+class CleanLibraries(CleanCommand):
+    @staticmethod
+    def remove_dynamic_library():
+        dst = os.path.join(HERE, os.path.join("ddtrace", "appsec", "ddwaf", "libddwaf"))
+        shutil.rmtree(dst, True)
```

Comment:
```suggestion
        shutil.rmtree(LIBDDWAF_DOWNLOAD_DIR, True)
```

iiuc this is the same correct?